core/modules/checkout/store/checkout/actions.ts

```diff
@@ -6,65 +6,74 @@ import { Logger } from '@vue-storefront/core/lib/logger'
-        await dispatch('cart/clear', { recreateAndSyncCart: true }, {root: true})
-        if (state.personalDetails.createAccount) {
-          commit(types.CHECKOUT_DROP_PASSWORD)
-        }
+        await dispatch('saveBypassTs')
+        await dispatch('cart/clear', { recreateAndSyncCart: true }, { root: true })
+        dispatch('dropPassword')
       }
     } catch (e) {
       Logger.error(e, 'checkout')()
     }
   },
+  async saveBypassTs () {
+    const userStorage = StorageManager.get('user')
+    await userStorage.setItem('last-cart-bypass-ts', new Date().getTime())
+  },
+  dropPassword ({ commit, state }) {
+    if (state.personalDetails.createAccount) {
+      commit(types.CHECKOUT_DROP_PASSWORD)
+    }
+  },
   setModifiedAt ({ commit }, timestamp) {
     commit(types.CHECKOUT_SET_MODIFIED_AT, timestamp)
   },
   savePersonalDetails ({ commit }, personalDetails) {
-    // todo: create and move perdonal details vuex
     commit(types.CHECKOUT_SAVE_PERSONAL_DETAILS, personalDetails)
   },
   saveShippingDetails ({ commit }, shippingDetails) {
-    // todo: move to shipping vuex
     commit(types.CHECKOUT_SAVE_SHIPPING_DETAILS, shippingDetails)
   },
   savePaymentDetails ({ commit }, paymentDetails) {
-    // todo: move to payment vuex
     commit(types.CHECKOUT_SAVE_PAYMENT_DETAILS, paymentDetails)
   },
-  load ({ commit }) {
-    StorageManager.get('checkout').getItem('personal-details', (err, details) => {
-      if (err) throw new Error(err)
-      if (details) {
-        commit(types.CHECKOUT_LOAD_PERSONAL_DETAILS, details)
-      }
-    })
-    StorageManager.get('checkout').getItem('shipping-details', (err, details) => {
-      if (err) throw new Error(err)
-      if (details) {
-        commit(types.CHECKOUT_LOAD_SHIPPING_DETAILS, details)
-      }
-    })
-    StorageManager.get('checkout').getItem('payment-details', (err, details) => {
-      if (err) throw new Error(err)
-      if (details) {
-        commit(types.CHECKOUT_LOAD_PAYMENT_DETAILS, details)
-      }
-    })
+  async load ({ commit }) {
+    const checkoutStorage = StorageManager.get('checkout')
+    const personalDetails = await checkoutStorage.getItem('personal-details')
+    const shippingDetails = await checkoutStorage.getItem('shipping-details')
+    const paymentDetails = await checkoutStorage.getItem('payment-details')
+
+    if (personalDetails) {
+      commit(types.CHECKOUT_LOAD_PERSONAL_DETAILS, personalDetails)
+    }
+
+    if (shippingDetails) {
+      commit(types.CHECKOUT_LOAD_SHIPPING_DETAILS, shippingDetails)
+    }
+
+    if (paymentDetails) {
+      commit(types.CHECKOUT_LOAD_PAYMENT_DETAILS, paymentDetails)
+    }
   },
   updatePropValue ({ commit }, payload) {
     commit(types.CHECKOUT_UPDATE_PROP_VALUE, payload)
   },
   setThankYouPage ({ commit }, payload) {
     commit(types.CHECKOUT_SET_THANKYOU, payload)
+  },
+  addPaymentMethod ({ commit }, paymentMethod) {
```

Comment:
please make these actions async also. We don't need it now, but as we established - all actions by default should be async. In future, we can easily add to them something without breaking changes.